plugin/plugin.go

```diff
@@ -0,0 +1,288 @@
+package plugin
+
+import (
+	"context"
+	"encoding/xml"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+)
+
+// Args represents the plugin's configurable arguments.
+type Args struct {
+	ReportFilenamePattern     string `envconfig:"PLUGIN_REPORT_FILENAME_PATTERN"`
+	FailedFails               int    `envconfig:"PLUGIN_FAILED_FAILS"`
+	FailedSkips               int    `envconfig:"PLUGIN_FAILED_SKIPS"`
+	FailureOnFailedTestConfig bool   `envconfig:"PLUGIN_FAILURE_ON_FAILED_TEST_CONFIG"`
+	UnstableFails             int    `envconfig:"PLUGIN_UNSTABLE_FAILS"`
+	UnstableSkips             int    `envconfig:"PLUGIN_UNSTABLE_SKIPS"`
+	JobStatus                 string `envconfig:"PLUGIN_JOB_STATUS"`
+	ThresholdMode             int    `envconfig:"PLUGIN_THRESHOLD_MODE"`
+	PluginFailIfNoResults     bool   `envconfig:"PLUGIN_FAIL_IF_NO_RESULTS"`
+	Level                     string `envconfig:"PLUGIN_LOG_LEVEL"`
+}
+
+// ValidateInputs ensures the user inputs meet the plugin requirements.
+func ValidateInputs(args Args) error {
+	if args.ReportFilenamePattern == "" {
+		return errors.New("missing required parameter: ReportFilenamePattern. Please specify the pattern to locate the TestNG report files")
+	}
+	if args.FailedFails < 0 || args.FailedSkips < 0 || args.UnstableFails < 0 || args.UnstableSkips < 0 {
+		return errors.New("threshold values must be non-negative. Check the configured values for failed and skipped tests")
+	}
+	if args.ThresholdMode != 1 && args.ThresholdMode != 2 {
+		return errors.New("invalid ThresholdMode value. It must be 1 (absolute) or 2 (percentage). Check the configuration")
+	}
+	return nil
+}
+
+// Exec handles TestNG XML report processing and logs details.
+func Exec(ctx context.Context, args Args) error {
+	files, err := locateFiles(args.ReportFilenamePattern)
+	if err != nil {
+		logger := logrus.WithError(err)
+		logger.Error("Error locating files")
+		return errors.New("failed to locate files: " + err.Error())
+	}
+
+	if len(files) == 0 {
+		if args.PluginFailIfNoResults {
+			return errors.New("no TestNG XML report files found. Check the report file pattern")
+		}
+		logrus.Warn("No TestNG XML report files found, continuing execution as PluginFailIfNoResults is false")
```

Comment:
lets remove PluginFailIfNoResults and just throw error when the report file isn't available